src/internal/group_api/mod.rs

```diff
@@ -307,37 +360,78 @@ pub fn group_create<'a, CR: rand::CryptoRng + rand::RngCore>(
     group_id: Option<GroupId>,
     name: Option<GroupName>,
     add_as_member: bool,
+    members: &'a Vec<UserId>,
     needs_rotation: bool,
-) -> impl Future<Item = GroupMetaResult, Error = IronOxideErr> + 'a {
-    transform::gen_group_keys(recrypt)
-        .and_then(move |(plaintext, group_priv_key, group_pub_key)| {
-            // encrypt the group secret to the current user as they will be the group admin
-            Ok({
-                let encrypted_group_key = recrypt.encrypt(
-                    &plaintext,
-                    &user_master_pub_key.into(),
-                    &auth.signing_private_key().into(),
-                )?;
-
-                // compute a transform key if we are adding the caller as a group member as well
-                let transform_key: Option<crate::internal::TransformKey> = if add_as_member {
-                    Some(
-                        recrypt
-                            .generate_transform_key(
-                                &group_priv_key.into(),
-                                &user_master_pub_key.into(),
-                                &auth.signing_private_key().into(),
-                            )?
-                            .into(),
-                    )
-                } else {
-                    None
-                };
-                (group_pub_key, encrypted_group_key, transform_key)
-            })
+) -> impl Future<Item = GroupCreateResult, Error = IronOxideErr> + 'a {
+    user_api::user_key_list(auth, members)
+        .and_then(move |member_ids_and_keys| {
+            // this will occur when one of the UserIds in the members list cannot be found
+            if member_ids_and_keys.len() != members.len() {
+                // figure out which user ids could not be found in the database and include them in the error message.
+                use std::{collections::HashSet, iter::FromIterator};
+                let desired_members_set: HashSet<&UserId> = HashSet::from_iter(members);
+                let found_members: Vec<UserId> =
+                    member_ids_and_keys.into_iter().map(|(x, _)| x).collect();
+                let found_members_set: HashSet<&UserId> = HashSet::from_iter(&found_members);
+                // TODO (reviewers): this double reference scares me, but seems to work? Is there a better way to do this?
```

Comment:
I don't have a better way to do it.

Reply:
Great, then I'll try to be less scared. :ghost: